rust/ql/lib/codeql/rust/dataflow/internal/SsaImpl.qll

```diff
@@ -426,3 +459,49 @@ class PhiReadNode extends DefinitionExt, Impl::PhiReadNode {
 
   override Location getLocation() { result = Impl::PhiReadNode.super.getLocation() }
 }
+
+private module DataFlowIntegrationInput implements Impl::DataFlowIntegrationInputSig {
+  class Expr extends CfgNodes::AstCfgNode {
+    predicate hasCfgNode(SsaInput::BasicBlock bb, int i) { this = bb.getNode(i) }
+  }
+
+  Expr getARead(Definition def) { result = Cached::getARead(def) }
+
+  /** Holds if SSA definition `def` assigns `value` to the underlying variable. */
+  predicate ssaDefAssigns(WriteDefinition def, Expr value) {
+    exists(BasicBlock bb, int i |
+      def.definesAt(_, bb, i) and
+      value.getAstNode() = bb.getNode(i).getAstNode().(AssignmentExpr).getRhs()
```

Comment:
I think this should simply by `value = bb.getNode(i)`. We should then add a disjunct to `localFlowStepCommon` which transfers data from the RHS in an `AssignmentExpr` to the LHS.

Reply:
With the latest change to `ssaDefAssigns` we are now getting data flow steps from the SSA. So that seems promising 😁